### ISSUE TITLE
Error when querying mediaItems if the page they were uploaded to has been trashed

### ISSUE THREAD

**Issue by CalebBarnes** [wp-graphql/wp-graphql#1146]
I'm using WP GraphQL 0.6.1 with gatsby-source-wordpress-experimental. 

I uploaded some images to a page, trashed the page, then added the images to a new page. After that, querying ancestors on mediaItems throws this error. This error only happens when not authorized (GraphQL Playground). There is no error when using the WP GraphiQL plugin.

query:
```
{
  mediaItems(first: 1) {
    nodes {
      ancestors {
        __typename
      }
    }
  }
}

```

response:
```
{
  "errors": [
    {
      "debugMessage": "Abstract type PostObjectUnion must resolve to an Object type at runtime for field MediaItem.ancestors with value \"instance of WPGraphQL\\Model\\Post\", received \"null\".Either the PostObjectUnion type should provide a \"resolveType\" function or each possible types should provide an \"isTypeOf\" function.",
      "message": "Internal server error",
      "category": "internal",
      "locations": [
        {
          "line": 4,
          "column": 7
        }
      ],
      "path": [
        "mediaItems",
        "nodes",
        0,
        "ancestors",
        0
      ]
    }
  ],
  "data": {
    "mediaItems": {
      "nodes": [
        {
          "ancestors": [
            null
          ]
        }
      ]
    }
  }
}
```

I've tested this while authed and the ancestor is indeed the trashed page. ID 720.

```
      {
          "altText": "",
          "ancestors": [
            {
              "__typename": "Page",
              "databaseId": 720,
              "id": "cGFnZTo3MjA=",
              "title": "Home",
              "isRestricted": null
            }
          ]
        },
```

**Comment by jasonbahl** [comment]
@CalebBarnes Thanks for reporting this! 

TL;DR;

- Convert ancestors field to a proper GraphQL Connection
- Resolve nodes earlier so those that don't pass through the Model layer can be stripped before returning (See #1138)

----

I think one of the first things we should do here is consider making this a connection instead of a `listOf` as there can be contextual data between media items and their ancestors, so providing the proper `ancestors.edges.nodes` shape with a place to expose edge data would make sense long term. 

Then, once this is a proper connection, I think this would be solved by the work being done to address #1138 

Right now in many cases, nodes are being resolved a bit late in the cycle and by time they pass through the model layer it's kind of "too late" and we end up throwing errors, when the proper behavior should be to strip the node that's been restricted by the model layer from the list of nodes before returning the list at all.

**Comment by jasonbahl** [comment]
This appears to have been fixed. I cannot replicate. 

![Screen Shot 2020-12-08 at 5 59 52 AM](https://user-images.githubusercontent.com/1260765/101486898-98c65600-391a-11eb-91fe-cb49a23924c2.png)
